app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt

```diff
@@ -80,9 +102,59 @@ class ServerViewModel @Inject constructor(
             ).addTo(compositeDisposable)
     }
 
-    override fun onCleared() {
-        compositeDisposable.dispose()
-        super.onCleared()
+    fun handleWebSocketServerBinder(binder: WebSocketServerService.Binder) {
+        Timber.d("handleWebSocketServerBinder($binder)")
+
+        compositeDisposable += binder.clientConnectionStatus().subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeBy(
+                onNext = { connectionStatus ->
+                    mutablePulsatingViewStatus.postValue(connectionStatus)
+                },
+                onError = { Timber.e(it) }
+            )
+
+        compositeDisposable += binder.messages()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { (ws, message) ->
+                message.action?.let {
+                    handleMessageAction(it)
+                }
+                message.pushNotificationsToken?.let {
+                    receiveFirebaseToken(ws.remoteSocketAddress.address.hostAddress, it)
+                }
+                message.babyName?.let { name ->
+                    mutableBabyNameStatus.postValue(name)
+                }
+                message.pairingCode?.let {
+                    mutablePairingCodeLiveData.postValue(it)
+                }
+            }
+    }
+
+    private fun handleMessageAction(action: String) {
+        when (action) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/b1aa35793dca6d7714e5534d6167592f0bbe6a99/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt#L137"></span>Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
  </tbody>
</table>